### PR TITLE
Quarkus docs contributor guide & template enhancements

### PR DESCRIPTION
**Pull request by michelle-purcell**
Some enhancements to the instructions for contributing to Quarkus documentation and to the templates for creating the Quarkus-styled _diataxis_ content types.

### CODE REVIEW

**Comment by michelle-purcell** [comment]
FYI: @MichalMaler , @sheilamjones , @rolfedh @inoxx03

**Review by rolfedh — APPROVED** [review]
Excellent updates. Thank you for doing this.

**Comment by MichalMaler** [comment]
Top notch stuff, @michelle-purcell

File: docs/src/main/asciidoc/doc-contribute-docs-howto.adoc [hunk 1]
```diff
@@ -82,22 +82,27 @@ Where:
...
+
+* For minor documentation changes, you can use the AsciiDoc syntax highlighting and preview provided by your IDE.
+* For significant changes or updates to generated configuration documentation, build the `docs` module locally and run the Vale linter as outlined in the following sections.
+
+=== Build the `docs` module locally
 
 include::{includes}/compile-quarkus-quickly.adoc[tag=quickly-install-docs]
 
-This will produce:
+Running `DquicklyDocs`` produces:
```
Comment (ebullient) [inline docs/src/main/asciidoc/doc-contribute-docs-howto.adoc:105]:
```suggestion
Running `-DquicklyDocs`` produces:
```

File: docs/src/main/asciidoc/doc-contribute-docs-howto.adoc [hunk 2]
```diff
@@ -124,10 +129,13 @@ When updating extension configuration:
...
 == Lint documentation changes with Vale
 
-Our builds use https://vale.sh[Vale] to check grammar, style, and word usage in the English versions of our documents.
-The configuration for vale is in the `docs/.vale` directory.
+Our builds use https://vale.sh[Vale] to check grammar, style, and word usage in the English versions of our documents. We created our own Vale style ruleset to ensure that content aligns with the preferred Quarkus style guidelines.
 
-NOTE: The Vale configuration file is `docs/.vale/vale.ini`.
+[NOTE]
+====
+* The Quarkus configuration for vale is `vale.ini`, which is in the `docs/.vale` directory.
```
Comment (ebullient) [inline docs/src/main/asciidoc/doc-contribute-docs-howto.adoc:136]: watch delivery of #29913 ... conflict here

**Review by ebullient — APPROVED** [review]

File: docs/src/main/asciidoc/doc-contribute-docs-howto.adoc (hunk 2 above)
Comment (michelle-purcell) [reply docs/src/main/asciidoc/doc-contribute-docs-howto.adoc:136]: @ebullient - Thanks for flagging. I'm now :eyes: subscribed to PR [29913](https://github.com/quarkusio/quarkus/pull/29913).

**Comment by michelle-purcell** [comment]
@ebullient, @MichalMaler , @rolfedh  - Thanks for your reviews. All comments are now resolved. This is ready to merge.

**Review by ebullient — APPROVED** [review]

**Pull request merged**